hdate/date.py

```diff
@@ -139,31 +196,40 @@ def _jdn(self) -> int:
 
     @property
     def hebrew_date(self) -> str:
-        """Return the hebrew date string."""
-        _month = cast(Months, self.hdate.month)
-        return (
-            f"{hebrew_number(self.hdate.day, hebrew=self.hebrew)} "  # Day
-            f"{htables.MONTHS[_month.value - 1][self.hebrew + 1]} "  # Month
-            f"{hebrew_number(self.hdate.year, hebrew=self.hebrew)}"  # Year
-        )
+        """Return the hebrew date string in the selected language."""
+        day = self.get_hebrew_number(self.hdate.day)
+        month_enum = cast(Months, self.hdate.month)
+        month = getattr(htables.MONTHS[month_enum.value - 1], self.language)
+        year = self.get_hebrew_number(self.hdate.year)
+        return f"{day} {month} {year}"
 
     @property
     def parasha(self) -> str:
-        """Return the upcoming parasha."""
-        return cast(str, htables.PARASHAOT[self.get_reading()][self.hebrew + 1])
+        """Return the upcoming parasha in the selected language."""
+        parasha_index = self.get_reading()
+        parasha = cast(str, getattr(htables.PARASHAOT[parasha_index], self.language))
+        return parasha
 
     @property
-    def holiday_description(self) -> str:
+    def holiday_description(self) -> Optional[str]:
         """
-        Return the holiday description.
-
-        In case none exists will return None.
+        Return the holiday description in the selected language.
+        If none exists, will return None.
         """
         entries = self._holiday_entries()
-        return ", ".join(
-            entry.description.hebrew.long if self.hebrew else entry.description.english
-            for entry in entries
-        )
+        descriptions = []
+        for entry in entries:
+            # Access the language-specific description
+            description_language = getattr(
+                HOLIDAY_DESCRIPTIONS.get(entry.name), self.language, None
+            )
+            if description_language:
+                # Check if it's a DESC namedtuple with 'long' and 'short' attributes
```

Comment:
Thanks for dealing with that. Although I do plan getting rid of short descriptions.